### DISCUSSION TITLE
ADML/ADMX creation

### DISCUSSION THREAD

**Discussion by KnifMelti** [Weatherlights/Winget-AutoUpdate-Intune#2] [Q&A] [answered]
Hi, @Weatherlights !
Did you create yours by hand (like I did for WAU) or have you found any editor anywhere?
I just can't imagine Microsoft, Google, Citrix etc., etc.  does this with copy and paste!
I still haven't found any...

**Comment by Weatherlights** [comment, marked as answer]
Hey @KnifMelti :)

There is a tool out there from FullArmor that was once available through Microsoft. But it can still be downloaded from 3rd Party Websites like this one: https://www.freedownloadmanager.org/landing.htm?f=fdmlib If you take a look in the Google ADMX you can see many traces that Google used this tool aswell :). But it does not cover all ADMX features.

However I started with the registry structure and then used a script out of my PolicyApplicator Conversion Kit which takes a given registry structure and creates an ADMX from it and the ADML and the rest I did by hand aswell (It is painfully to debug isn't it? :D).

There is also another tool out there to dump you registry into an ADMX file which will - unlike like my tool - also create an ADMX file: https://call4cloud.nl/2022/10/memoirs-of-an-admx/

**Reply by KnifMelti** [reply]
Thank you!
